azure-mgmt-containerinstance/src/main/java/com/microsoft/azure/management/containerinstance/implementation/ContainerImpl.java

```diff
@@ -0,0 +1,230 @@
+ * license information.
+ */
+package com.microsoft.azure.management.containerinstance.implementation;
+
+import com.microsoft.azure.management.apigeneration.LangDefinition;
+import com.microsoft.azure.management.containerinstance.Container;
+import com.microsoft.azure.management.containerinstance.ContainerGroup;
+import com.microsoft.azure.management.containerinstance.ContainerGroupNetworkProtocol;
+import com.microsoft.azure.management.containerinstance.ContainerPort;
+import com.microsoft.azure.management.containerinstance.EnvironmentVariable;
+import com.microsoft.azure.management.containerinstance.IpAddress;
+import com.microsoft.azure.management.containerinstance.Port;
+import com.microsoft.azure.management.containerinstance.ResourceRequests;
+import com.microsoft.azure.management.containerinstance.ResourceRequirements;
+import com.microsoft.azure.management.containerinstance.VolumeMount;
+
+import java.util.ArrayList;
+import java.util.Map;
+
+/**
+ * Implementation for container group's container instance definition stages interface.
+ */
+@LangDefinition
+class ContainerImpl implements
+    ContainerGroup.DefinitionStages.ContainerInstanceDefinitionStages.ContainerInstanceDefinition<ContainerGroup.DefinitionStages.WithNextContainerInstance> {
+    private Container innerContainer;
+    private ContainerGroupImpl parent;
+
+    ContainerImpl(ContainerGroupImpl parent, String containerName) {
+        this.parent = parent;
+        this.innerContainer = new Container()
+            .withName(containerName)
+            .withResources(new ResourceRequirements()
+                .withRequests(new ResourceRequests()
+                    .withCpu(1)
+                    .withMemoryInGB(1.5)));
+    }
+
+    @Override
+    public ContainerGroupImpl attach() {
+        if (parent.inner().containers() == null) {
+            parent.inner().withContainers(new ArrayList<Container>());
+        }
+        parent.inner().containers().add(innerContainer);
+
+        return parent;
+    }
+
+    @Override
+    public ContainerImpl withImage(String imageName) {
+        innerContainer.withImage(imageName);
+        return this;
+    }
+
+    @Override
+    public ContainerImpl withoutPorts() {
+        innerContainer.withPorts(null);
+
+        return this;
+    }
+
+    @Override
+    public ContainerImpl withExternalTcpPorts(int... ports) {
+        for (int port : ports) {
+            this.withExternalTcpPort(port);
+        }
+
+        return this;
+    }
+
+    @Override
+    public ContainerImpl withExternalTcpPort(int port) {
+        ensureParentIpAddress().ports().add(new Port()
+            .withPort(port)
+            .withProtocol(ContainerGroupNetworkProtocol.TCP));
+        this.withInternalPort(port);
+
+        return this;
+    }
+
+    private IpAddress ensureParentIpAddress() {
+        if (parent.inner().ipAddress() == null) {
+            parent.inner().withIpAddress(new IpAddress()
+                .withType("Public")
```

Comment:
"Public"? So they don't provide an enum constant for this?

Reply:
no enums are available :-(